bokeh/document.py

```diff
@@ -752,10 +752,6 @@ def validate(self):
             None
 
         '''
-        # logging.basicConfig is a no-op if there's already
-        # some logging configured. We want to make sure warnings
-        # go somewhere so configure here if nobody has.
-        logging.basicConfig(level=logging.INFO)
```

Comment:
Calling `basicConfig` in random library functions is completely unpredictable (and broken from the user's POV, as whether bokeh messes with the logging configuration then depends on which exact library functions are called). logging should either be configured at library startup, or shouldn't be configured at all.